tool/net/lfuncs.c

```diff
@@ -602,6 +602,64 @@ int LuaEncodeLatin1(lua_State *L) {
   }
 }
 
+int tobits(int b) {
+  int bits = 0; while (b && (b >>= 1)) bits++;
+  return bits;
+}
+
+// the next function is based on
+// https://github.com/google/google-authenticator-libpam/blob/master/src/base32.c
+// Copyright 2010 Google Inc.; Author: Markus Gutschke
```

Comment:
Could you put it in its own file, with its own license header?

Reply:
@jart, I reorganized all the code and added decoding (along with the documentation).